src/php/lib/Grpc/BaseStub.php

```diff
@@ -196,17 +211,184 @@ private function _validate_and_normalize_metadata($metadata)
+                $metadata = call_user_func(
+                    $this->update_metadata,
+                $metadata,
+                $jwt_aud_uri
+                );
+            }
+            $metadata = $this->_validate_and_normalize_metadata(
+                $metadata
+            );
+            $call->start($argument, $metadata, $options);
+            return $call;
+        };
+    }
+
+    private function GrpcStreamUnary($channel)
+    {
+        return function ($method, $deserialize, array $metadata = [], array $options = []) use ($channel) {
+            $call = new ClientStreamingCall(
+                $channel,
+                $method,
+                $deserialize,
+                $options
+            );
+            $jwt_aud_uri = $this->_get_jwt_aud_uri($method);
+            if (is_callable($this->update_metadata)) {
+                $metadata = call_user_func(
+                    $this->update_metadata,
+                $metadata,
+                $jwt_aud_uri
+                );
+            }
+            $metadata = $this->_validate_and_normalize_metadata(
+                $metadata
+            );
+            $call->start($metadata);
+            return $call;
+        };
+    }
+
+    private function GrpcUnaryStream($channel)
+    {
+        return function ($method, $argument, $deserialize, array $metadata = [], array $options = []) use ($channel) {
+            $call = new ServerStreamingCall(
+                $channel,
+                $method,
+                $deserialize,
+                $options
+            );
+            $jwt_aud_uri = $this->_get_jwt_aud_uri($method);
+            if (is_callable($this->update_metadata)) {
+                $metadata = call_user_func(
+                    $this->update_metadata,
+                $metadata,
+                $jwt_aud_uri
+                );
+            }
+            $metadata = $this->_validate_and_normalize_metadata(
+                $metadata
+            );
+            $call->start($argument, $metadata, $options);
+            return $call;
+        };
+    }
+
+    private function GrpcStreamStream($channel)
+    {
+        return function ($method, $deserialize, array $metadata = [], array $options = []) use ($channel) {
+            $call = new BidiStreamingCall(
+                $channel,
+                $method,
+                $deserialize,
+                $options
+            );
+            $jwt_aud_uri = $this->_get_jwt_aud_uri($method);
+            if (is_callable($this->update_metadata)) {
+                $metadata = call_user_func(
+                    $this->update_metadata,
+                $metadata,
+                $jwt_aud_uri
+                );
+            }
+            $metadata = $this->_validate_and_normalize_metadata(
+                $metadata
+            );
+            $call->start($metadata);
+
+            return $call;
+        };
+    }
+
+    private function UnaryUnaryCallFactory($channel)
```

Comment:
Can a subclass accidentally override a private method in PHP by declaring a method of the same name? In that case, we probably need to ensure it is preceded by underscore, as I read in the comment string for `_simpleRequest` below.